vault/logical_system_raft.go

```diff
@@ -335,6 +382,116 @@ func (b *SystemBackend) handleStorageRaftSnapshotRead() framework.OperationFunc
 	}
 }
 
+func (b *SystemBackend) handleStorageRaftAutopilotHealth() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		raftBackend, ok := b.Core.underlyingPhysical.(*raft.RaftBackend)
+		if !ok {
+			return logical.ErrorResponse("raft storage is not in use"), logical.ErrInvalidRequest
+		}
+
+		health, err := raftBackend.GetAutopilotServerHealth(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		if health == nil {
+			return nil, nil
+		}
+
+		return &logical.Response{
+			Data: map[string]interface{}{
+				"healthy":                      health.Healthy,
+				"failure_tolerance":            health.FailureTolerance,
+				"optimistic_failure_tolerance": health.OptimisticFailureTolerance,
+				"servers":                      health.Servers,
+				"leader":                       health.Leader,
+				"voters":                       health.Voters,
+				"read_replicas":                health.ReadReplicas,
+			},
+		}, nil
+	}
+}
+
+func (b *SystemBackend) handleStorageRaftAutopilotConfigRead() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		raftStorage, ok := b.Core.underlyingPhysical.(*raft.RaftBackend)
+		if !ok {
+			return logical.ErrorResponse("raft storage is not in use"), logical.ErrInvalidRequest
+		}
+		config := raftStorage.AutopilotConfig()
```

Comment:
Can this be nil?  Do we want to give a different response if disabled?

Reply:
It can be nil. Fixed. API will silently return nothing.